code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -51,20 +81,17 @@
 	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(handle_stealth))
 	RegisterSignal(owner, COMSIG_XENOMORPH_POUNCE_END, PROC_REF(sneak_attack_pounce))
 	RegisterSignal(owner, COMSIG_XENO_LIVING_THROW_HIT, PROC_REF(mob_hit))
+	if(pounce_hit_break_stealth)
+		RegisterSignal(owner, COMSIG_XENOMORPH_LEAP_BUMP, PROC_REF(mob_hit))
 	RegisterSignal(owner, COMSIG_XENOMORPH_ATTACK_LIVING, PROC_REF(sneak_attack_slash))
 	RegisterSignal(owner, COMSIG_XENOMORPH_DISARM_HUMAN, PROC_REF(sneak_attack_slash))
 	RegisterSignal(owner, COMSIG_XENOMORPH_ZONE_SELECT, PROC_REF(sneak_attack_zone))
 	RegisterSignal(owner, COMSIG_XENOMORPH_PLASMA_REGEN, PROC_REF(plasma_regen))
 
 	// TODO: attack_alien() overrides are a mess and need a lot of work to make them require parentcalling
-	RegisterSignals(owner, list(
-		COMSIG_XENOMORPH_GRAB,
-		COMSIG_XENOMORPH_THROW_HIT,
-		COMSIG_LIVING_IGNITED,
-		COMSIG_LIVING_ADD_VENTCRAWL,
-		COMSIG_XENOABILITY_MIRAGE_SWAP), PROC_REF(cancel_stealth))
-
-	RegisterSignal(owner, COMSIG_XENOMORPH_ATTACK_OBJ, PROC_REF(on_obj_attack))
+	RegisterSignals(owner, disable_on_signals, PROC_REF(cancel_stealth))
+	if(disable_on_obj_slash)
+		RegisterSignal(owner, COMSIG_XENOMORPH_ATTACK_OBJ, PROC_REF(on_obj_attack))
```

Comment:
Being able to just slash cades/turrets while maintaining stealth so turrets can literally never hit you seems uh... questionable.

You are also making this way more complicated than it needs to be. The bool var should be a flag, but it doesn't need to exist at all. Just use child procs that do what you want to do (i.e. in this case, nothing)

Reply:
So.. I should do nothing?